### ISSUE TITLE
navigate to byte position

### ISSUE THREAD

**Issue by kurtbuilds** [zed-industries/zed#19327]
### Check for existing issues

- [X] Completed

### Describe the feature

when debugging binary files, sometimes the error message given is a byte position, not a line & column number. it'd be awesome to have a cmd+k command to jump to a byte position in a file to identify the error.

**Comment by notpeter** [comment]
Yeah, if we were supporting byte offset, it might also be appropriate to support UTF-8 character position as well.  Given that your external tool might say "character 1234" which actually mapped to "byte 1255" or some such.

This could conceivably be added to `GotoLine` (`ctrl-g`) with a prefix. So like `@123` for byte offset.

Could you also paste the output of your tool that provided a byte position?  I ask because we currently support hyperlinking `filename:3:5` to `filename line 3 column 5` with a number of different formats.  We could conceivably add support for jumping directly to the position in question.

- See also https://github.com/zed-industries/zed/issues/18952 for the way VSCode handles those URLs.  We could conceivable also support an `offset` field for ProblemMatcher style regexes.
